mat/cholesky_test.go

```diff
@@ -707,3 +708,219 @@ func BenchmarkCholeskyInverseTo(b *testing.B) {
 		})
 	}
 }
+
+func TestBandCholeskySolveTo(t *testing.T) {
+	t.Parallel()
+
+	const (
+		nrhs = 4
+		tol  = 1e-14
+	)
+	rnd := rand.New(rand.NewSource(1))
+	for _, n := range []int{1, 2, 3, 5, 10} {
+		for _, k := range []int{0, 1, n / 2, n - 1} {
+			k := min(k, n-1)
+
+			a := NewSymBandDense(n, k, nil)
+			for i := 0; i < n; i++ {
+				a.SetSymBand(i, i, rnd.Float64()+float64(n))
+				for j := i + 1; j < min(i+k+1, n); j++ {
+					a.SetSymBand(i, j, rnd.Float64())
```

Comment:
Do we want to test with internal/testrand?

Reply:
I've thought about it but in my view testrand is most useful for testing numerical properties/robustness which I think should be done in the lapack package. Testing in mat should focus on the added value/functionality of mat which is the handling of variety of matrix types. Do you agree?

I've never used testrand but it looks like it could be used more in testlapack to ease (after some extending) the generation of random test matrices.

Reply:
Yes, I agree.